lib/oidc/getWithPopup.ts

```diff
@@ -33,3 +33,47 @@ export function getWithPopup(sdk: OktaAuthOAuthInterface, options: TokenParams):
   });
   return getToken(sdk, options);
 }
+
+export function getWithIDPPopup(sdk: OktaAuthOAuthInterface, options: TokenParams): { cancel: () => void, promise: Promise<TokenResponse> } {
+  if (!BroadcastChannel) {
+    throw new AuthSdkError('Modern browser with `BroadcastChannel` support is required to use this method');
+  }
+
+  if (!options.state) {
+    options.state = generateState();
+  }
+
+  // some browsers (safari, firefox) block popup if it's initialed from an async process
+  // here we create the popup window immediately after user interaction
+  // then redirect to the /authorize endpoint when the requestUrl is available
+  const popupWindow = loadPopup('/', options);
+  const channel = new BroadcastChannel(`popup-callback:${options.state}`);
+
+  options = clone(options) || {};
+  Object.assign(options, {
+    display: 'popup',
+    responseMode: 'query',
+    popupWindow,
+    idpPopup: true,
+    channel,
+  });
+
+  let cancelPromise;
+  const promise = new Promise<TokenResponse>((resolve, reject) => {
+    cancelPromise = reject;
+    return getToken(sdk, options)
+    .then((res) => {
+      return resolve(res);
+    });
+  });
+
+  const cancel = () => {
+    channel.close();
+    cancelPromise(new AuthSdkError('Popup flow canceled'));
```

Comment:
All of the Errors are `AuthSdkError` -- would we expect host applications to need to catch and handle these errors normally? If so, would it help to differentiate them with some specific error types that extend `AuthSdkError` so they are easier to catch? Otherwise, you need to check the `error.message`, which is more brittle